src/com/mrpowergamerbr/powersuperantiproxy/listeners/AsyncJoinListener.java

```diff
@@ -25,7 +26,7 @@ public AsyncJoinListener(PowerSuperAntiProxy m) {
         this.m = m;
     }
 
-    @EventHandler
+    @EventHandler(priority = EventPriority.HIGHEST)

```

Comment:
Se for para colocar HIGHEST, então também coloque ignoreCancelled = true para não gastar processamento com usuários que já foram kickados por outro motivo.